core/src/main/java/de/javakaffee/web/msm/MemcachedElasticConnectionFactory.java

```diff
@@ -0,0 +1,108 @@
+package de.javakaffee.web.msm;
+
+import net.spy.memcached.BinaryConnectionFactory;
+import net.spy.memcached.ConnectionFactory;
+import net.spy.memcached.DefaultConnectionFactory;
+
+import java.lang.reflect.*;
+
+/**
+ * Created by asodhi on 11/29/2016.
+ */
+public class MemcachedElasticConnectionFactory implements StorageClientFactory.MemcachedNodeURLBasedConnectionFactory {
+
+    @Override
+    public ConnectionFactory createBinaryConnectionFactory(final long operationTimeout,
+                                                           final long maxReconnectDelay,
+                                                           boolean isClientDynamicMode) {
+        try {
+            Class clientMode = Class.forName("net.spy.memcached.ClientMode");
```

Comment:
elasticache-java-cluster-client and java-memcached-client have method signature which are common but do different things.
net.spy.memcached.BinaryConnectionFactory() would create a standard connection with memcached if java-memcached-client is loaded first same method would create a dynamic connection with aws if elasticache-java-cluster-client is loaded first.